File: Data Structures & Algorithms/split-array-largest-sum/submission-0.go
import ("slices")
func splitArray(nums []int, k int) int {
	if k == len(nums) {
		return slices.Max(nums)
	}
	left := slices.Max(nums)
	right:= 0
	for _, value := range nums {
		right += value
	}
	for left < right {
		mid := left + (right - left) / 2



		if isMaxSplit(nums, k, mid) {
			right = mid
		} else {
			left = mid+1
		}
	 
	}
	return left
}

func isMaxSplit(nums []int, k int, max int) bool {
	count := 1
	currentSum := 0

	for i:= 0; i<len(nums); i++ {
		currentSum += nums[i]
		if currentSum > max {
			currentSum = nums[i]
			count++
		}
	}
	return count <= k
}

File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        if len(nums) == 2 and nums[0] != nums[1]:
            return False
        
        elif len(nums) < 2:
            return False

        for i in range(0, len(nums)):
            for j in range(i+1, len(nums)):
                if nums[i] == nums[j]:
                    return True
        return False